tardis/visualization/tools/tests/test_sdec_plot.py

```diff
@@ -0,0 +1,511 @@
+            group.create_dataset("_species_name", data=plotter._species_name)
+            group.create_dataset("_color_list", data=plotter._color_list)
+
+            fig_subgroup = group.create_group("fig_data")
+
+            for index, data in enumerate(fig.get_children()):
+                trace_group = fig_subgroup.create_group(str(index))
+
+                if isinstance(data.get_label(), str):
+                    trace_group.attrs["name"] = data.get_label()
+
+                # save artists which correspond to element contributions
+                if isinstance(data, PolyCollection):
+                    for index, path in enumerate(data.get_paths()):
+                        trace_group.create_dataset(
+                            "path" + str(index),
+                            data=path.vertices,
+                        )
+                # save line plots
+                if isinstance(data, Line2D):
+                    trace_group.create_dataset("data", data=data.get_xydata())
+                    trace_group.create_dataset(
+                        "path", data=data.get_path().vertices
+                    )
+
+        else:
+            group = self.hdf_file[subgroup_name]
+            # test output of the _make_colorbar_labels function
+            assert (
+                plotter._species_name
+                == group.get("_species_name").asstr()[()].tolist()
+            )
+            # test output of the _make_colorbar_colors function
+            np.testing.assert_allclose(
+                np.asarray(np.asarray(plotter._color_list)),
+                group.get("_color_list")[()],
+            )
+            fig_subgroup = group.get("fig_data")
+            for index, data in enumerate(fig.get_children()):
+                trace_group = fig_subgroup.get(str(index))
+
+                if isinstance(data.get_label(), str):
+                    assert trace_group.attrs["name"] == data.get_label()
+
+                # test element contributions
+                if isinstance(data, PolyCollection):
+                    for index, path in enumerate(data.get_paths()):
+                        np.testing.assert_allclose(
+                            trace_group.get("path" + str(index))[()],
+                            path.vertices,
+                        )
+                # compare line plot data
+                if isinstance(data, Line2D):
+                    np.testing.assert_allclose(
+                        trace_group.get("data")[()], data.get_xydata()
+                    )
+                    np.testing.assert_allclose(
+                        trace_group.get("path")[()],
+                        data.get_path().vertices,
+                    )
+
+    @pytest.mark.parametrize("packets_mode", ["virtual", "real"])
+    @pytest.mark.parametrize("packet_wvl_range", [[500, 9000] * u.AA])
+    @pytest.mark.parametrize("distance", [10 * u.Mpc, 50 * u.Mpc])
```

Comment:
I believe another param for distance should be None like in mpl?

Reply:
Done!

Reply:
Done!